x-pack/plugins/maps/public/components/layer_addpanel/view.js

```diff
@@ -4,167 +4,131 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { Component, Fragment } from 'react';
-import { ALL_SOURCES } from '../../shared/layers/sources/all_sources';
+import React, { Component } from 'react';
+import { SourceSelect } from './source_select/source_select';
+import { FlyoutFooter } from './flyout_footer';
+import { SourceEditor } from './source_editor/view';
 import {
-  EuiButton,
-  EuiButtonEmpty,
   EuiFlexGroup,
-  EuiFlexItem,
   EuiTitle,
-  EuiPanel,
-  EuiSpacer,
-  EuiCard,
-  EuiIcon,
   EuiFlyoutHeader,
-  EuiFlyoutFooter,
 } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import _ from 'lodash';
 
 export class AddLayerPanel extends Component {
 
   state = {
     sourceType: null,
-    layer: null
+    layer: null,
+    indexingTriggered: false,
+    indexingComplete: false,
+    importIndexingReady: false,
+    importView: false,
   }
 
-  _previewLayer = (source) => {
+  _viewLayer = (source, fitToExtent = false) => {
     if (!source) {
       this.setState({ layer: null });
       this.props.removeTransientLayer();
       return;
     }
 
-
     const layerOptions = this.state.layer
       ? { style: this.state.layer.getCurrentStyle().getDescriptor() }
       : {};
+    const layer = source.createDefaultLayer(layerOptions, this.props.mapColors);
+    this.setState({ layer }, async () => {
+      await this.props.viewLayer(this.state.layer);
+      fitToExtent && this.props.fitToLayerExtent(layer.getId());
+    });
+  };
+
+  _addImportedLayer = async source => {
+    await this.props.removeTransientLayer();
```

Comment:
should flag the corresponding `props.removeTransientLayer` with `async` for symmetry if we're using `await` here